node/node.go

```diff
@@ -245,12 +245,23 @@ func NewNode(config *cfg.Config,
 		state.ConsensusParams.BlockSize.MaxBytes,
 		state.Validators.Size(),
 	)
+	maxGas := state.ConsensusParams.BlockSize.MaxGas
 	mempool := mempl.NewMempool(
 		config.Mempool,
 		proxyApp.Mempool(),
 		state.LastBlockHeight,
 		mempl.WithMetrics(memplMetrics),
-		mempl.WithFilter(func(tx types.Tx) bool { return len(tx) <= maxDataBytes }),
+		mempl.WithPreCheckFilter(func(tx types.Tx) bool {
+			// We have to account for the amino overhead in the tx size as well
+			aminoOverhead := amino.UvarintSize(uint64(len(tx)))
+			return (len(tx) + aminoOverhead) <= maxDataBytes
```

Comment:
This seems to big of a core concern to have it linger in the setup code of the system. This function should be implemented in the package itself and harnessed with tests there and then be used in the setup of the mempool. It could look like this:

``` go
// AminoMaxBytes needs to be documented.
func AminoMaxBytes(maxDataBytes int) PreCheckFilterFunc {
	return func(tx types.Tx) bool {
		// We have to account for the amino overhead in the tx size as well
		aminoOverhead := amino.UvarintSize(uint64(len(tx)))
		return (len(tx) + aminoOverhead) <= maxDataBytes
	}
}

// ...
	mempl.WithPreCheckFilter(mempl.AminoMaxBytes(maxDataBytes))
// ...
```

Reply:
Our mempool is an interface, with the interface defined in `state`. It seems a bit weird for `AminoMaxBytes` to be defined in the state package. Is it fine to introduce the actual mempool package as an import here for `AminoMaxBytes`?

Reply:
The `func AminoMaxBytes` was meant to go into the `mempool` package.

Reply:
What I mean is that if `AminoMaxBytes` was in the mempool, `state` would have to import `mempool`. This is kind of weird, since `state` also defines the `mempool` interface.

Reply:
How does the implementation of `AminoMaxBytes` in the `mempool` package necessitate the import of `mempool` in `state`? It's not part of the mempool interface.